src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -558,7 +558,7 @@ private SupportedPlatform GetCurrentRunningOS()
                     // So disable it, but don't complain that the rule was explicitly disabled.
                     disabledSkimmers.Add(skimmer.Id);
                 }
-                else if (skimmer.DefaultConfiguration.Level != failureLevel)
+                else if (skimmer.DefaultConfiguration.Level != failureLevel && ruleEnabled != RuleEnabledState.Default && ruleEnabled != RuleEnabledState.Disabled)
```

Comment:
>skimmer [](start = 25, length = 7)

style issue, can you make this a multiline check, not put it on a single line? thanks!